src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/ToplevelBuilder.java

```diff
@@ -177,8 +175,17 @@ String mods() {
         @Override
         void classBegin() {
             super.classBegin();
-            emitLibraries(libraryNames);
             emitConstructor();
+            // emit basic primitive types
+            emitPrimitiveTypedef(Type.primitive(Type.Primitive.Kind.Bool), "C_BOOL");
+            emitPrimitiveTypedef(Type.primitive(Type.Primitive.Kind.Char), "C_CHAR");
+            emitPrimitiveTypedef(Type.primitive(Type.Primitive.Kind.Short), "C_SHORT");
+            emitPrimitiveTypedef(Type.primitive(Type.Primitive.Kind.Int), "C_INT");
+            emitPrimitiveTypedef(Type.primitive(Type.Primitive.Kind.Long), "C_LONG");
+            emitPrimitiveTypedef(Type.primitive(Type.Primitive.Kind.LongLong), "C_LONG_LONG");
+            emitPrimitiveTypedef(Type.primitive(Type.Primitive.Kind.Float), "C_FLOAT");
+            emitPrimitiveTypedef(Type.primitive(Type.Primitive.Kind.Double), "C_DOUBLE");
+            emitPointerTypedef("C_POINTER");
         }
```

Comment:
Will this clash with user's typedef?

typedef int C_INT; 

in user's header?

Reply:
possibly, perhaps we can introduce some disambiguation later?

Reply:
sounds good